src/overlays/actors/ovl_En_Guard_Nuts/z_en_guard_nuts.c

```diff
@@ -28,42 +36,338 @@ const ActorInit En_Guard_Nuts_InitVars = {
     (ActorFunc)EnGuardNuts_Draw,
 };
 
-// static ColliderCylinderInit sCylinderInit = {
-static ColliderCylinderInit D_80ABBDF0 = {
-    { COLTYPE_NONE, AT_NONE, AC_NONE, OC1_ON | OC1_TYPE_PLAYER, OC2_TYPE_2, COLSHAPE_CYLINDER, },
-    { ELEMTYPE_UNK0, { 0xF7CFFFFF, 0x00, 0x00 }, { 0xF7CFFFFF, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_NONE, OCELEM_ON | OCELEM_UNK3, },
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_NONE,
+        OC1_ON | OC1_TYPE_PLAYER,
+        OC2_TYPE_2,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_NONE,
+        OCELEM_ON | OCELEM_UNK3,
+    },
     { 50, 50, 0, { 0, 0, 0 } },
 };
 
-#endif
+s32 D_80ABBE1C = 0;
+
+s32 D_80ABBE20 = 0;
+
+static u16 sTextIDs[] = { 0x0824, 0x0825, 0x0826, 0x082D, 0x0827, 0x0828, 0x0829, 0x082A, 0x082B, 0x082C };
+
+s16 D_80ABBE38[] = { 0x0000, 0x0000, 0x0002, 0x0001, 0x0000, 0x0000, 0x0002, 0x0000, 0x0000, 0x0002 };
+
+static AnimationHeader* sAnimations[] = {
+    &gDekuPalaceGuardWaitAnim,
+    &gDekuPalaceGuardWalkAnim,
+    &gDekuPalaceGuardDigAnim,
+    &gDekuPalaceGuardWalkAnim,
+};
+
+static u8 sAnimModes[] = { ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_ONCE, ANIMMODE_ONCE };
+
+static TexturePtr sEyeTextures[] = {
+    gDekuPalaceGuardEyeOpenTex,
+    gDekuPalaceGuardEyeHalfTex,
+    gDekuPalaceGuardEyeClosedTex,
+};
+
+void EnGuardNuts_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnGuardNuts* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 20.0f);
+    SkelAnime_Init(globalCtx, &this->skelAnime, &gDekuPalaceGuardSkel, &gDekuPalaceGuardWaitAnim, this->jointTable,
+                   this->morphTable, OBJECT_DNK_LIMB_MAX);
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    this->actor.targetMode = 1;
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    Actor_SetScale(&this->actor, 0.015f);
+    Math_Vec3f_Copy(&this->unk228, &this->actor.world.pos);
+    this->unk21E = D_80ABBE1C;
+    D_80ABBE1C++;
```

Comment:
Hmm... looks like its counting how many guard_nuts are loaded, saving their number in order they were initialized... but its unused (by guard nuts anyway, maybe another actor would look for it)

Reply:
Oh uh... I meant more along the lines of the .data value being the count, and the saved value of unk21E being an identifier.

Like, in order:
[The First Actor] will init and the starting value of (0) is saved to that actor, then data is incremented to (1)
[The Second Actor] will init and the starting value will be (1) from the last actor, the unk21E will be (1) because it was incremented from the last actor, and then we increment D_80ABBE1C to 2, which will be the identifier of the next actor (except there isn't a third actor)

I mean obviously its not used, so its intentional purpose isn't really known as it wasn't fully implemented in the final game, but it looks like it was a way to specify one guard compared to the other from load order rather than params. Maybe they wanted the two guards to talk to each other and wanted to identify tweedle dee from tweedle dumb but changed their mind